src/main/java/de/bonndan/nivio/assessment/kpi/KPIFactory.java

```diff
@@ -32,16 +29,18 @@ public KPIFactory(ApplicationEventPublisher eventPublisher) {
     /**
      * Returns the default KPIs plus custom ones and configuration.
      *
-     * @param landscape the landscape
+     * @param kpiConfigMap the landscape kpi configurations
      * @return effective KPIs
+     * @throws ProcessingException if a KPI is misconfigured
      */
-    public Map<String, KPI> getConfiguredKPIs(Landscape landscape) {
+    public Map<String, KPI> getConfiguredKPIs(@NonNull Map<String, KPIConfig> kpiConfigMap) {
 
-        Map<String, KPIConfig> config = landscape.getConfig().getKPIs();
-        Map<String, KPI> kpis = new HashMap<>(defaultKPIs.size() + config.size());
+        Objects.requireNonNull(kpiConfigMap, "kpi config is null or empty");
```

Comment:
`Objects.requireNonNull()` does not check if the map is empty, only if an object == null.

Reply:
Yes. If the are no configured KPIs, the map would be empty.

Reply:
I meant that the message is missleading, we are not checking if the map is empty, only if it is null ;)

Reply:
Fixed in a different branch.